src/Language/PureScript/TypeChecker.hs

```diff
@@ -434,31 +434,116 @@ typeCheckModule (Module ss coms mn decls (Just exps)) =
   warnAndRethrow (addHint (ErrorInModule mn)) $ do
     modify (\s -> s { checkCurrentModule = Just mn })
     decls' <- typeCheckAll mn exps decls
+    classesToSuperClasses <- gets
+      ( M.map
+        ( S.fromList
+        . fmap constraintClass
```

Comment:
Rather than performing the intersections below to make sure we only keep local superclasses*, can we do a `filter` here which inspects the `Constraint` values and only keeps the superclasses which have been defined in the current module? That way, if I understand correctly, we won't need to construct the `moduleClassDeclarations` map (which saves us from iterating over all of the declarations in the module).

\*that is the reason for the intersections below, right?

Reply:
okay, that's a much better idea :joy: